pyproject.toml

```diff
@@ -32,7 +32,10 @@ maintainers = [
 name = "pydistcheck"
 readme = "README.md"
 requires-python = ">=3.8"
-version = "0.4.0.99"
+dynamic = ["version"]
+
+[tool.setuptools.dynamic]
+version = {attr = "pydistcheck.__version__"}
```

Comment:
reference on this: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html